src/datepicker-input/datepicker-input.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, Input } from "@angular/core";
+
+let nextId = 0;
```

Comment:
This should be a static property on the class ... `static private datePickerInputCount = 0;` for example